File: tech-studio-projects/financial-industry/algo-trading-platform/java/algo-trading-platform/src/com/fintech/algotrading/orders/OrderCore.java
package com.fintech.algotrading.orders;

public class OrderCore {
    private final long orderId;
    private final String username;
    private final int securityId;

    // Constructor implementation.
    public OrderCore(long orderId, String username, int securityId) {
        this.orderId = orderId;
        this.username = username;
        this.securityId = securityId;
    }

    // Getter for the Order ID.
    public long getOrderId() {
        return orderId;
    }

    // Getter for the Username.
    public String getUsername() {
        return username;
    }

    // Getter for the Security ID.
    public int getSecurityId() {
        return securityId;
    }
}

File: tech-studio-projects/financial-industry/algo-trading-platform/cpp/orders/OrderCore.cpp
#include "OrderCore.h"

// Constructor implementation.
OrderCore::OrderCore(long orderId, const std::string& username, int securityId)
    : OrderId(orderId), Username(username), SecurityId(securityId) {
    // Initialization is handled in the member initializer list.
}

// Getter for the Order ID.
long OrderCore::GetOrderId() const {
    return OrderId;
}

// Getter for the Username.
std::string OrderCore::GetUsername() const {
    return Username;
}

// Getter for the Security ID.
int OrderCore::GetSecurityId() const {
    return SecurityId;
}
